benchmarks/benchmarks/analysis/distances.py

```diff
@@ -11,120 +17,236 @@
         """Benchmark calculation of all distances
         within a single numpy array of coordinates
         using self_distance_array with preallocated
         result array.
         """
         distances.self_distance_array(reference=self.coords_1,
-                                      box=None,
+                                      box=self.box_dims,
                                       result=self.allocated_array_1D,
                                       backend='serial')
 
-    def time_contact_matrix(self, num_atoms):
-        """Benchmark calculation of contacts within
-        a single numpy array using the default arguments
-        to contact_matrix.
-        """
-        distances.contact_matrix(coord=self.coords_1,
-                                 cutoff=15.0,
-                                 returntype='numpy',
-                                 box=None)
-
-    def time_contact_matrix_sparse(self, num_atoms):
-        """Benchmark calculation of contacts within
-        a single numpy array using the slower reduced
-        memory implementation of contact_matrix intended
-        for larger systems.
-        """
-        distances.contact_matrix(coord=self.coords_1,
-                                 cutoff=15.0,
-                                 returntype='sparse',
-                                 box=None)
-
-    def time_dist(self, num_atoms):
+    def time_dist(self, num_atoms, box_dims):
         """Benchmark calculation of distances between
         atoms in two atomgroups with no offsets
         to resids.
         """
         distances.dist(A=self.ag1,
                        B=self.ag2,
+                       box=self.box_dims,
                        offset=0)
 
-    def time_dist_offsets(self, num_atoms):
+    def time_dist_offsets(self, num_atoms, box_dims):
         """Benchmark calculation of distances between
         atoms in two atomgroups with offsets
         to resids.
         """
         distances.dist(A=self.ag1,
                        B=self.ag2,
+                       box=self.box_dims,
                        offset=20)
 
-    def time_between(self, num_atoms):
-        """Benchmark determination of subgroup
-        of atomgroup that is within a specific
-        distance of two other atomgroups.
+    def time_contact_matrix(self, num_atoms, pbc_type):
+        """Benchmark calculation of contacts within
+        a single numpy array using the default arguments
+        to contact_matrix.
         """
-        distances.between(group=self.ag3,
-                          A=self.ag1,
-                          B=self.ag2,
-                          distance=15.0)
+        distances.contact_matrix(coord=self.ag1.positions,
+                                 cutoff=15.0,
+                                 returntype='numpy',
+                                 box=self.box_dims)
+
+    def time_contact_matrix_sparse(self, num_atoms, box_dims):
```

Comment:
You have benchmark functions in the same class with different parameters, but normally the same parameters are used for each. In fact, it is extra confusing when I look at the output (see below) because `box_dims` gets replaced by the true parameter name (`pbc_type`) it seems, which would not be very useful for an analysis of a performance regression (a user/developer would have to realise that an override/hack has happened, if that's what is happening here...).

Running this locally:

```
[ 12.50%] ··· analysis.distances.DistancesBench.time_contact_matrix_sparse                                                                                                                                                                                                                                                ok
[ 12.50%] ··· =========== ========== ============ ===========
              --                        pbc_type             
              ----------- -----------------------------------
               num_atoms     None     orthogonal   triclinic 
              =========== ========== ============ ===========
                   10      466±0μs     473±0μs      425±0μs  
                  100      33.3±0ms    34.5±0ms     35.1±0ms 
                  1000     1.51±0s     2.81±0s      2.47±0s  
                 10000     25.9±0s     35.4±0s      27.6±0s  
              =========== ========== ============ ===========
```

Reply:
See commit [d8d42c8](https://github.com/MDAnalysis/mdanalysis/pull/3475/commits/d8d42c8a3add4d984a15bcfde522efbf453ebd2c)

Took me a bit to figure out what you meant but I completely agree that this is a clearer way of writing it. Please let me know of any other suggestions you have.